redfish-core/lib/hypervisor_system.hpp

```diff
@@ -167,77 +168,117 @@ inline bool extractHypervisorInterfaceData(
-                                     ethIfaceId + "/ipv4/addr0")
+            std::pair<boost::container::flat_set<IPv6AddressData>::iterator,
+                      bool>
+                v6Itr = ipv6Config.insert(IPv6AddressData{});
+
+            IPv4AddressData& ipv4Address = *v4Itr.first;
+            IPv6AddressData& ipv6Address = *v6Itr.first;
+
+            for (std::string protocol : {"ipv4", "ipv6"})
             {
-                idFound = true;
-                if (interface.first == "xyz.openbmc_project.Network.IP")
+                std::string ipAddrObj =
+                    "/xyz/openbmc_project/network/hypervisor/";
+                ipAddrObj.append(ethIfaceId);
+                ipAddrObj.append("/");
+                ipAddrObj.append(protocol);
+                ipAddrObj.append("/addr0");
+                if (objpath.first == ipAddrObj)
                 {
-
-                    for (auto& property : interface.second)
+                    idFound = true;
+                    if (interface.first == "xyz.openbmc_project.Network.IP")
                     {
-                        if (property.first == "Address")
+
+                        for (const auto& property : interface.second)
                         {
-                            const std::string* address =
-                                std::get_if<std::string>(&property.second);
-                            if (address != nullptr)
+                            if (property.first == "Address")
                             {
-                                ipv4Address.address = *address;
+                                const std::string* address =
+                                    std::get_if<std::string>(&property.second);
+                                if (address != nullptr)
+                                {
+                                    if (protocol == "ipv4")
+                                    {
+                                        ipv4Address.address = *address;
+                                    }
+                                    else if (protocol == "ipv6")
+                                    {
+                                        ipv6Address.address = *address;
+                                    }
+                                }
                             }
-                        }
-                        else if (property.first == "PrefixLength")
-                        {
-                            const uint8_t* mask =
-                                std::get_if<uint8_t>(&property.second);
-                            if (mask != nullptr)
+                            else if (property.first == "PrefixLength")
                             {
-                                ipv4Address.netmask = getNetmask(*mask);
+                                const uint8_t* mask =
+                                    std::get_if<uint8_t>(&property.second);
+                                if (mask != nullptr)
```

Comment:
bmcweb now does 
                             if (mask== nullptr)
                             {
                              internalError; 
                              return

Reply:
Same comments below for gateway and enabled

Reply:
Done